src/wallet/rpcdump.cpp

```diff
@@ -151,9 +151,17 @@ UniValue importprivkey(const JSONRPCRequest& request)
         CKeyID vchAddress = pubkey.GetID();
         {
             pwallet->MarkDirty();
-            // We don't know which corresponding address will be used; label them all
+
+            // We don't know which corresponding address will be used;
+            // label all new addresses, and label existing addresses if a
+            // label was passed.
             for (const auto& dest : GetAllDestinationsForKey(pubkey)) {
-                pwallet->SetAddressBook(dest, strLabel, "receive");
+                const auto& it = pwallet->mapAddressBook.find(dest);
```

Comment:
Nit: It's slightly confusing to use a reference type here; `find` returns an iterator by value.